pkg/apis/core/validation/cloudprofile.go

```diff
@@ -325,17 +325,23 @@ func validateMachineImages(machineImages []core.MachineImage, fldPath *field.Pat
 			}
 
 			allErrs = append(allErrs, validateExpirableVersion(machineVersion.ExpirableVersion, helper.ToExpirableVersions(image.Versions), versionsPath)...)
-			allErrs = append(allErrs, validateContainerRuntimesInterfaces(machineVersion.CRI, versionsPath.Child("cri"))...)
+			allErrs = append(allErrs, validateContainerRuntimesInterfaces(machineVersion.CRI, image, versionsPath.Child("cri"))...)
 			allErrs = append(allErrs, validateMachineImageVersionArchitecture(machineVersion.Architectures, versionsPath.Child("architecture"))...)
 		}
 	}
 
 	return allErrs
 }
 
-func validateContainerRuntimesInterfaces(cris []core.CRI, fldPath *field.Path) field.ErrorList {
+func validateContainerRuntimesInterfaces(cris []core.CRI, image core.MachineImage, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	duplicateCRI := sets.String{}
+	hasDocker := false
+
+	if len(cris) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath, fmt.Sprintf("must provide at least one supported container runtime for machine image %q", image.Name)))
```

Comment:
`image.Name` is not fully qualifying, as this is validated per version. I think you can simply remove it because the error will already contain the full field path.